### PR TITLE
Task #223691 feat: Updated - Edit, Add new in district component and fixed UT issues, Refactored 

### PR DESCRIPTION
**Pull request by rahulg1254**

### CODE REVIEW

File: src/components/AddBlockModal.tsx [hunk 1]
```diff
@@ -59,92 +60,156 @@ export const AddBlockModal: React.FC<AddBlockModalProps> = ({
...
-      errors.value = t("COMMON.INVALID_TEXT");
-      valid = false;
-    }
+  useEffect(() => {
+    const fetchDistricts = async () => {
+      try {
+        const response = await getDistrictsForState({
+          limit: 10,
+          offset: 0,
+          controllingfieldfk: "MH",
```
Comment (itsvick) [inline src/components/AddBlockModal.tsx:72]: @rahulg1254 why are u hardcoding it here

File: src/components/AddBlockModal.tsx [hunk 2]
```diff
@@ -59,92 +60,156 @@ export const AddBlockModal: React.FC<AddBlockModalProps> = ({
...
-      valid = false;
-    } else if (!isAlphabetic(formData.value)) {
-      errors.value = t("COMMON.INVALID_TEXT");
-      valid = false;
-    }
+  useEffect(() => {
+    const fetchDistricts = async () => {
+      try {
+        const response = await getDistrictsForState({
+          limit: 10,
```
Comment (itsvick) [inline src/components/AddBlockModal.tsx:70]: @rahulg1254 keep this in the app.constant file

File: src/components/AddBlockModal.tsx [hunk 3]
```diff
@@ -59,92 +60,156 @@ export const AddBlockModal: React.FC<AddBlockModalProps> = ({
...
 
-    if (!formData.controllingField.trim()) {
-      errors.controllingField = t("COMMON.REQUIRED_FIELD");
-      valid = false;
-    } else if (!isAlphabetic(formData.controllingField)) {
-      errors.controllingField = t("COMMON.INVALID_TEXT");
-      valid = false;
-    }
+        if (response) {
+          setDistricts(response.result.values);
```
Comment (itsvick) [inline src/components/AddBlockModal.tsx:77]: @rahulg1254 use nullish cohesion operator here

File: src/components/AddBlockModal.tsx [hunk 4]
```diff
@@ -59,92 +60,156 @@ export const AddBlockModal: React.FC<AddBlockModalProps> = ({
...
+        setDistricts([]);
+      }
+    };
 
-    setErrors(errors);
-    return valid;
+    if (open) fetchDistricts();
+  }, [open, districtId]);
+  const validateField = (
+    field: keyof typeof formData,
```
Comment (itsvick) [inline src/components/AddBlockModal.tsx:91]: @rahulg1254 what are u trying to here

File: src/components/AddBlockModal.tsx [hunk 5]
```diff
@@ -59,92 +60,156 @@ export const AddBlockModal: React.FC<AddBlockModalProps> = ({
...
+        [field]: errorMessage,
+      }));
+    };
+
+  const validateForm = () => {
+    const newErrors = {
+      name: validateField(
+        "name",
+        formData.name,
+        t("COMMON.BLOCK_NAME_REQUIRED")
```
Comment (itsvick) [inline src/components/AddBlockModal.tsx:140]: @rahulg1254 looks like duplicated

File: src/pages/state.tsx [hunk 6]
```diff
@@ -238,7 +240,7 @@ const State: React.FC = () => {
             }))}
             limit={pageLimit}
             offset={pageOffset}
-            paginationEnable={true}
+            paginationEnable={paginationCount >= 5}
```
Comment (itsvick) [inline src/pages/state.tsx:243]: @rahulg1254 add this default size in app.constant

**Review by itsvick — CHANGES_REQUESTED** [review]

File: src/pages/state.tsx (hunk 6 above)
Comment (rahulg1254) [reply src/pages/state.tsx:243]: Resolved

File: src/components/AddBlockModal.tsx (hunk 4 above)
Comment (rahulg1254) [reply src/components/AddBlockModal.tsx:93]: resolved

File: src/components/AddBlockModal.tsx (hunk 5 above)
Comment (rahulg1254) [reply src/components/AddBlockModal.tsx:142]: resolved

File: src/components/AddBlockModal.tsx (hunk 3 above)
Comment (rahulg1254) [reply src/components/AddBlockModal.tsx:78]: resolved

File: src/components/AddBlockModal.tsx (hunk 2 above)
Comment (rahulg1254) [reply src/components/AddBlockModal.tsx:71]: resolved

File: src/components/AddBlockModal.tsx (hunk 1 above)
Comment (rahulg1254) [reply src/components/AddBlockModal.tsx]: resolved

**Pull request merged**